_posts/2020-09-10-status.md

```diff
@@ -0,0 +1,55 @@
+---
+title: "Abseil Status"
+layout: blog
+sidenav: side-nav-blog.html
+published: true
+permalink: blog/2020-091021-status  
+type: markdown
+category: blog
+excerpt_separator: <!--break-->
+---
+
+By [Xiaoyi Zhang](mailto:zhangxy988@gmail.com), Google Engineer, Emeritus
+
+The Abseil status library is now available on abseil.io. This library is used
+within Google for error handling and contains the following two abstractions:
+
+* `absl::Status`
+* `absl::StatusOr<T>` 
+
+Within Google, `absl::Status` is the primary mechanism to gracefully handle
+errors across API boundaries (and in particular across RPC boundaries). Some of
+these errors may be recoverable, but others may not. Most functions which can
+produce a recoverable error should be designed to return either an
+absl::Status (or the similar `absl::StatusOr<T>`, which holds either an object
+of type `T` or an error).
+
+<!--break-->
+
+A Status can either return “OK” (represented by the `enum absl::StatusCode:kOk`)
+or one of a number of canonical error codes to indicate certain error conditions: 
+
+```cpp
+absl::Status MyFunction(absl::string_view fname, ...) {
```

Comment:
s/fname/name/ (or "filename") here?  `fname` is really not a, um, name that we want people using.